internal/core/metadata/v2/application/device.go

```diff
@@ -194,6 +198,8 @@ func PatchDevice(dto dtos.UpdateDevice, ctx context.Context, dic *di.Container)
 		correlation.FromContext(ctx),
 	))
 
+	dto.Name = &device.Name
+	go updateDeviceCallback(ctx, dic, dto)
```

Comment:
pass the full content DTO for the update callback

Reply:
if the device service is changed, we have to call the callback of the original and new device service.
@iain-anderson you have made this in V1.  I am wondering we should change the way in V2.
you call the two Device Service "Update" callback respectively.  I thought we can call "Add" callback to the new Device Service, and call "Delete" callback to the original Device Service.  It would impact the implementation in Device SDK.  What do you think?

Reply:
@cloudxxx8 I see what you mean in that the required actions from the two services are equivalent to delete and add, but I prefer "update" as it conveys more information to the device service. So for example the original DS can log that the the device has moved to another service, rather than just giving a "deleted" message.

Reply:
@cloudxxx8 I see what you mean in that the required actions from the two services are equivalent to delete and add, but I prefer "update" as it conveys more information to the device service. So for example the original DS can log that the the device has moved to another service, rather than just giving a "deleted" message.
We talked about this in the WG and I think we don't want to do too much work in this area until we've got a use case where devices are being migrated. When we try to do a real-world implementation we might rethink how best to make it all work.

Reply:
> pass the full content DTO for the update callback

Modified.

> but I prefer "update" as it conveys more information to the device service.

Invoke new and old service callback.